### PR TITLE
Select LogAnalytics region based on OSA region

### PR DESCRIPTION
**Pull request by jim-minter**
@pweil- @amanohar ptal

### CODE REVIEW

**Comment by pweil-** [comment]
Question: do we need to provide this as `previous vendored commit` + this?  Are you comfortable with the merge log from then until now (looked ok to me for a revendor).

File: pkg/api/validate.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func Validate(new, old *OpenShiftManagedCluster, externalOnly bool) (errs []erro
 }
 
 func validateContainerService(c *OpenShiftManagedCluster, externalOnly bool) (errs []error) {
-	if c.Location == "" {
+	if _, found := AzureLocations[c.Location]; !found {
 		errs = append(errs, fmt.Errorf("invalid location %q", c.Location))
```
Comment (amanohar) [inline pkg/api/validate.go]: nit: It's better to say: 
> location not supported
since the location is valid.

**Comment by jim-minter** [comment]
@pweil- branching at 5b63224421c158caeae2cf0841537c3edab60756 and cherry-picking

**Comment by jim-minter** [comment]
nits updated, @pweil- please label lgtm

**Comment by pweil-** [comment]
> @pweil- branching at [5b63224](https://github.com/openshift/openshift-azure/commit/5b63224421c158caeae2cf0841537c3edab60756) and cherry-picking

10-4, thanks for handling this quickly

**Pull request merged**